mono/metadata/process.c

```diff
@@ -391,22 +382,25 @@ process_get_assembly_fileversion (MonoObject *filever, MonoAssembly *assembly)
 }
 
 static MonoObject*
-get_process_module (MonoAssembly *assembly, MonoClass *proc_class)
+get_process_module (MonoAssembly *assembly, MonoClass *proc_class, MonoError *error)
 {
-	MonoError error;
 	MonoObject *item, *filever;
 	MonoDomain *domain = mono_domain_get ();
 	char *filename;
 	const char *modulename = assembly->aname.name;
 
-	filename = g_strdup_printf ("[In Memory] %s", modulename);
+	mono_error_init (error);
 
 	/* Build a System.Diagnostics.ProcessModule with the data.
 	 */
-	item = mono_object_new_checked (domain, proc_class, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
-	filever = mono_object_new_checked (domain, mono_class_get_file_version_info_class (), &error);
-	if (!mono_error_ok (&error)) goto leave;
+	item = mono_object_new_checked (domain, proc_class, error);
+	if (!mono_error_ok (error))
+		return NULL;
```

Comment:
Also `return_val_if_nok (error, value)` that expands to `if (!mono_error_ok (error)) return value`